### ISSUE TITLE
Learning rate scheduler not in effect when using `interval='steps'`

### ISSUE THREAD

**Issue by GuillaumeLeclerc** [PyTorchLightning/pytorch-lightning#9475]
## 🐛 Bug

Trying to use Learning rate scheduler

### To Reproduce

https://colab.research.google.com/drive/1kn5RiRG6VH1TiOJY35d9ZjMobpyNP05K?usp=sharing

### Expected behavior

Learning rate is divided by 10 after 1000 steps of optimization. But instead it stays the same.

### Environment

```bash
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0+cu101
	- pytorch-lightning: 0.10.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020
```

**Comment by NielsRogge** [comment]
See #3869. You should probably also add 'frequency'.

I ran into the same issue, this should be better documented imo. cc @williamFalcon

**Comment by tchaton** [comment]
Hey @GuillaumeLeclerc,

It should be `interval=step` and not `interval=steps`, I believe this could be typo.

Mind trying ?

Best,
T.C

**Comment by rohitgr7** [comment]
the configuration is incorrect. pass it like:
```py
def configure_optimizers(self):
    optimizer = torch.optim.SGD(self.layer.parameters(), lr=0.1)
    self.oo = optimizer
    sched_config = {
        'scheduler': ch.optim.lr_scheduler.StepLR(optimizer, 10, 0.1),
        'frequency': 100,
        'interval': 'step'
    }
    return {
        'optimizer': optimizer,
        'lr_scheduler': sched_config
    }
```
more info here: https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#configure-optimizers

**Comment by rohitgr7** [comment]
@NielsRogge info regarding the frequency and other scheduler keys are mentioned in the docs here: https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#configure-optimizers

if you think it can be explained in a better way, do send a PR :)

**Comment by tchaton** [comment]
Dear @GuillaumeLeclerc,

Any updates ?

Best,
T.C

**Comment by GuillaumeLeclerc** [comment]
@tchaton It is working now in the test but still not in my real code. I'll investigate the differences now and reopen if I can't get it to work. Thanks!
